### PR TITLE
Added hotkeys in config.ini and keys in editor.cpp

### PR DESCRIPTION
**Pull request by tigres810**
List of changes I made:
config.ini
  Added hotkeys with detailed info how to use them.
Editor.cpp
  Added function helpers[toUpper, ContainsKey, HasKeyOnlyCTRL, HasKeyOnlySHIFT, HasKeyCTRLAndSHIFT, GetFirstNonModifierKey, StringToButton] all of these located at the top.
Replaced about text with config keys.
[Update] = [Added screenshot key. Added deselect all key. Added delete button key. Added camera movement. Added focus on selection key. Added inspector key. Added depth of field focus key. Added ragdoll tester key. Added colorgrading key. Added wireframe key. Added enable transform tool key. Added save key. Added select all entitys key. Added[copy, cut, paste, duplicate, put instances, undo, redo, editor toggles] keys and added some comments.]

### CODE REVIEW

File: Editor/Editor.cpp [hunk 1]
```diff
@@ -6,6 +6,11 @@
 #include "Translator.h"
 #include "DummyVisualizer.h"
 
+#include <string>
```
Comment (turanszkij) [inline Editor/Editor.cpp:9]: these includes should go to stdafx.h if they are not already there

File: Editor/Editor.cpp [hunk 2]
```diff
@@ -45,6 +50,79 @@ static wi::unordered_map<std::string, FileType> filetypes = {
 	{"H", FileType::HEADER},
 };
 
+// Hotkey functions
+std::string toUpper( std::string str )
```
Comment (turanszkij) [inline Editor/Editor.cpp:54]: This is already implemented in `wi::helper::toUpper`

File: Editor/Editor.cpp [hunk 3]
```diff
@@ -1221,12 +1302,12 @@ void EditorComponent::Update(float dt)
 			if (!wi::input::Down(wi::input::KEYBOARD_BUTTON_LCONTROL))
 			{
 				// Only move camera if control not pressed
-				if (wi::input::Down((wi::input::BUTTON)'A') || wi::input::Down(wi::input::GAMEPAD_BUTTON_LEFT)) { moveNew += XMVectorSet(-1, 0, 0, 0); }
-				if (wi::input::Down((wi::input::BUTTON)'D') || wi::input::Down(wi::input::GAMEPAD_BUTTON_RIGHT)) { moveNew += XMVectorSet(1, 0, 0, 0); }
-				if (wi::input::Down((wi::input::BUTTON)'W') || wi::input::Down(wi::input::GAMEPAD_BUTTON_UP)) { moveNew += XMVectorSet(0, 0, 1, 0); }
-				if (wi::input::Down((wi::input::BUTTON)'S') || wi::input::Down(wi::input::GAMEPAD_BUTTON_DOWN)) { moveNew += XMVectorSet(0, 0, -1, 0); }
-				if (wi::input::Down((wi::input::BUTTON)'E') || wi::input::Down(wi::input::GAMEPAD_BUTTON_2)) { moveNew += XMVectorSet(0, 1, 0, 0); }
-				if (wi::input::Down((wi::input::BUTTON)'Q') || wi::input::Down(wi::input::GAMEPAD_BUTTON_1)) { moveNew += XMVectorSet(0, -1, 0, 0); }
+				if (wi::input::Down(StringToButton(hotkeyssection.GetText("move_camera_left"))) || wi::input::Down(wi::input::GAMEPAD_BUTTON_LEFT)) { moveNew += XMVectorSet(-1, 0, 0, 0); }
```
Comment (turanszkij) [inline Editor/Editor.cpp:1305]: Doing string based lookups at runtime every frame is not good, and especially if it always tries to read from the config, something better has to be thought out, for example using enum of editor actions with a lookup table.

File: Editor/Editor.cpp [hunk 4]
```diff
@@ -1864,61 +1945,190 @@ void EditorComponent::Update(float dt)
...
-		if (wi::input::Press((wi::input::BUTTON)'C') || wi::input::Press((wi::input::BUTTON)'X'))
+
+		// Copy
+		std::string togglecopykey = hotkeyssection.GetText("copy_action");
+		bool istogglecopypressed = wi::input::Press(StringToButton(togglecopykey));
+
+		auto copyAction = [&]()
 		{
-			auto& prevSel = translator.selectedEntitiesNonRecursive;
+			auto & prevSel = translator.selectedEntitiesNonRecursive;
```
Comment (turanszkij) [inline Editor/Editor.cpp:2078]: This kind of formatting is not what the code uses, this commit shouldn't change just formatting of existing code.

File: Editor/Editor.cpp (hunk 1 above)
Comment (tigres810) [reply Editor/Editor.cpp:9]: I've checked the original source and #include "Translator.h"
#include "DummyVisualizer.h" are already in Editor.cpp should I still change them or only the string include to stdafx.h?

**Comment by turanszkij** [comment]
This looks pretty good now (without testing yet). I fixed the issue with the 1 key being broken and this breaks the PR slightly because the enum order changed and you added a new enum, please try to merge it if you have time (in wiInput.h and wiInput_BindLua.cpp). I will test it later.

**Review by tigres810 — COMMENTED** [review]
Changes made not reviewed

**Comment by turanszkij** [comment]
I'll review only after it builds

**Review by turanszkij — APPROVED** [review]

**Comment by turanszkij** [comment]
Thanks for the work.

**Pull request merged**